src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -88,53 +96,100 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             }
         }
 
         return settings;
     }
 
+    void TerminalSettings::_ApplyAppearanceSettings(const IAppearanceConfig& appearance, const Windows::Foundation::Collections::IMapView<winrt::hstring, ColorScheme>& schemes)
+    {
+        _CursorShape = appearance.CursorShape();
+        _CursorHeight = appearance.CursorHeight();
+        if (!appearance.ColorSchemeName().empty())
+        {
+            if (const auto scheme = schemes.TryLookup(appearance.ColorSchemeName()))
+            {
+                ApplyColorScheme(scheme);
+            }
+        }
+        if (appearance.Foreground())
+        {
+            _DefaultForeground = til::color{ appearance.Foreground().Value() };
+        }
+        if (appearance.Background())
+        {
+            _DefaultBackground = til::color{ appearance.Background().Value() };
+        }
+        if (appearance.SelectionBackground())
+        {
+            _SelectionBackground = til::color{ appearance.SelectionBackground().Value() };
+        }
+        if (appearance.CursorColor())
+        {
+            _CursorColor = til::color{ appearance.CursorColor().Value() };
+        }
+        if (!appearance.BackgroundImagePath().empty())
+        {
+            _BackgroundImage = appearance.ExpandedBackgroundImagePath();
+        }
+
+        _BackgroundImageOpacity = appearance.BackgroundImageOpacity();
+        _BackgroundImageStretchMode = appearance.BackgroundImageStretchMode();
+        std::tie(_BackgroundImageHorizontalAlignment, _BackgroundImageVerticalAlignment) = ConvertConvergedAlignment(appearance.BackgroundImageAlignment());
+
+        _RetroTerminalEffect = appearance.RetroTerminalEffect();
+        _PixelShaderPath = winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(appearance.PixelShaderPath().c_str()) };
+    }
+
     // Method Description:
     // - Creates a TerminalSettings object that inherits from a parent TerminalSettings
     // Arguments::
     // - parent: the TerminalSettings object that the newly created TerminalSettings will inherit from
     // Return Value:
     // - a newly created child of the given parent object
-    Model::TerminalSettings TerminalSettings::CreateWithParent(const Model::TerminalSettings& parent)
+    Model::TerminalSettingsStruct TerminalSettings::CreateWithParent(const Model::TerminalSettingsStruct& parent)
     {
         THROW_HR_IF_NULL(E_INVALIDARG, parent);
 
-        auto parentImpl{ get_self<TerminalSettings>(parent) };
-        return *parentImpl->CreateChild();
+        auto defaultImpl{ get_self<TerminalSettings>(parent.DefaultSettings()) };
+        auto defaultChild = defaultImpl->CreateChild();
+        if (parent.UnfocusedSettings())
+        {
+            parent.UnfocusedSettings().SetParent(*defaultChild);
+        }
+        return winrt::make<TerminalSettingsStruct>(*defaultChild, parent.UnfocusedSettings());

```

Comment:
Should the Unfocused settings inherit from the `defaultChild`, or should it inherit from parent's unfocused appearance? Can you break down why/which?

Reply:
Going to reply to this and the comment below here:

This function takes the pair of settings objects (instead of just one) only because we need to update the parentage of the second settings object. So, we actually return exactly the second settings object that was passed in, just with an updated parent. Its new parent is now the child of the first settings object passed in (`defaultChild`). 

In the spec we laid out that these unfocused settings objects inheritance should look like -
(this is the unfocused settings) -> term control overrides -> the profile that defined the unfocused settings

The term control overrides will live in `defaultChild`, and `defaultChild` has the 'OG' profile as its parent, so having `defaultChild` as the parent of the unfocused settings gets us that inheritance we wanted